src/components/rule-builder/delete-conditional-logic.modal.tsx

```diff
@@ -0,0 +1,269 @@
+import { ModalHeader, ModalBody, ModalFooter, Button } from '@carbon/react';
+import { useTranslation } from 'react-i18next';
+import { v4 as uuid } from 'uuid';
+import { showSnackbar } from '@openmrs/esm-framework';
+import { type Action, type Condition, type FormRule } from './rule-builder.component';
+import { type Schema } from '@types';
+import { findQuestionIndices } from '../utils';
+
+interface DeleteConditionalLogicModalProps {
+  closeModal: () => void;
+  questionId?: string;
+  questionLabel: string;
+  ruleId?: string;
+  currentRule: FormRule;
+  rules?: Array<FormRule>;
+  schema: Schema;
+  onSchemaChange: (schema: Schema) => void;
+  handleAddLogic?: (fieldId: string) => void;
+  setConditions?: React.Dispatch<React.SetStateAction<Array<Condition>>>;
+  setActions?: React.Dispatch<React.SetStateAction<Array<Action>>>;
+  setCurrentRule?: React.Dispatch<React.SetStateAction<FormRule>>;
+  setRules: React.Dispatch<React.SetStateAction<Array<FormRule>>>;
+  deleteAll?: boolean; // Flag to determine if all conditional logic should be deleted
+  setRuleDeleted: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const DeleteConditionalLogicModal: React.FC<DeleteConditionalLogicModalProps> = ({
+  closeModal,
+  questionId,
+  questionLabel,
+  currentRule,
+  schema,
+  onSchemaChange,
+  handleAddLogic,
+  setConditions,
+  setActions,
+  setCurrentRule,
+  setRules,
+  rules,
+  deleteAll = false,
+  setRuleDeleted,
+}) => {
+  const { t } = useTranslation();
+
+  const deleteQuestionHideProperty = useCallback(
+    (schema: Schema, pageIndex: number, sectionIndex: number, questionIndex: number) => {
+      const updatedSchema = { ...schema };
+      delete updatedSchema.pages[pageIndex].sections[sectionIndex].questions[questionIndex].hide;
+      onSchemaChange(updatedSchema);
+      return updatedSchema;
+    },
+    [onSchemaChange],
+  );
+
+  const deleteValidatorProperty = useCallback(
+    (schema: Schema, pageIndex: number, sectionIndex: number, questionIndex: number, errorMessage: string) => {
+      if (!errorMessage) return;
+      const updatedSchema = { ...schema };
+      const validatorIndex = currentRule?.validatorIndex;
+      const question = updatedSchema.pages[pageIndex].sections[sectionIndex].questions[questionIndex];
+      const validators = question.validators;
+      const questionId = question.id;
+      const existingValidator =
+        validatorIndex >= 1
+          ? updatedSchema.pages[pageIndex].sections[sectionIndex].questions[questionIndex].validators[
+              validatorIndex - 1
+            ]
+          : undefined;
+      const existingValidatorIndex = validators.findIndex((validator) => validator === existingValidator);
+      if (existingValidatorIndex !== -1) {
+        validators?.splice(existingValidatorIndex, 1);
+      }
+      setRules((prevRules) => {
+        const updatedRules = prevRules.map((rule) => {
+          const updatedRule = { ...rule };
```

Comment:
I don't think this is necessary. You can safely modify `prevRules` without negative effects.